### PR TITLE
Automate releases

### PR DESCRIPTION
**Pull request by pmeinhardt**
Attempt number 4. 😅 

**UPDATE**

We can now automatically upload to Chrome Web Store and Mozilla Add-Ons. Plus: we also create a GitHub release with automated notes.

**TODO**

The branch protection on the `main` branch may still be an issue for the automation...

### CODE REVIEW

File: script/list-changes [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
```
Comment (pmeinhardt) [inline script/list-changes:1]:
```suggestion
#!/usr/bin/env bash
```

**Comment by pmeinhardt** [comment]
Next time we release, let's take this for another test ride and merge if its okay.
We probably should update the README before merging.

**Comment by pmeinhardt** [comment]
Looks like signing with `web-ext` now works – cf. https://github.com/bitcrowd/tickety-tick/pull/249#issuecomment-652050512. Note: I only tried for unlisted versions so far.

```
yarn build:firefox && WEB_EXT_API_KEY=xxx WEB_EXT_API_SECRET=xxx yarn sign
yarn run v1.22.17
$ web-ext sign --source-dir ./dist/firefox --artifacts-dir ./dist --channel=unlisted
Applying config file: ./package.json
Building web extension from /Users/pmeinhardt/Code/bitcrowd/tickety-tick/dist/firefox
Validating add-on [...................................................................................]
Validation results: https://addons.mozilla.org/en-US/developers/upload/xxx
Signing add-on [......................................................................................]
Downloading signed files: 100%
Downloaded:
    dist/tickety_tick-5.0.0unlisted-an+fx.xpi
Extension ID: jid1-ynkvezs8Qn2TJA@jetpack
SUCCESS
✨  Done in 273.78s.
```

**Comment by klappradla** [comment]
> Looks like signing with `web-ext` now works – cf. [#249 (comment)](https://github.com/bitcrowd/tickety-tick/pull/249#issuecomment-652050512). Note: I only tried for unlisted versions so far.

Wow 🤯

**Comment by klappradla** [comment]
Small update: [web-ext-sign](https://extensionworkshop.com/documentation/develop/web-ext-command-reference/#web-ext-sign) has a flag to use the new submission API now.

File: script/release [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+# Build artifacts.
+yarn bundle:chrome
+yarn bundle:firefox
+
+# # Create a GitHub release, upload build artifacts.
+notes=$(./script/list-changes "$prev")
+tag=$(git describe --abbrev=0 --tags)
+gh release create "$tag" \
+  --notes "$notes" \
+  ./dist/*.zip
```
Comment (klappradla) [inline script/release:26]: `gh create` has an an interactive mode which allows to generate the release notes or use the ones on the git tag. By skipping `-- notes`, we could make use of that 🤔

**Comment by klappradla** [comment]
Went for a two step process now @pmeinhardt, in order to align with other bitcrowd open source projects and to not limit releasing to admins only:

1. PR with version bump (`yarn prepare-release`)
2. Actual release pipeline (`yarn release`)

This still yields the change of tagging the _"wrong"_ commit with the version. But in practice, we can assume that steps 1 and 2 are pretty much directly after each other and there won't be much happening aside that.

What do you think?

**Comment by pmeinhardt** [comment]
> Went for a two step process now @pmeinhardt, in order to align with other bitcrowd open source projects and to not limit releasing to admins only […]
> 
> What do you think?

Sure thing. You are the ones cutting the releases, so the process has to work for you. ✌️

**Review by maltoe — APPROVED** [review]
Nice work! :100:

**Pull request merged**